test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -256,10 +256,53 @@ upgrade_downgrade()
         done
 }
 
+#############################################################
+# test_upgrade_to_branch:
+#       arg1: release branch name
+#       arg2: path to test data folder
+#       arg3: path to test test root folder
+#############################################################
+test_upgrade_to_branch()
+{
+        cd $1/test/checkpoint
+
+        for FILE in $2/*; do
+            # Run actual test.
+            echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+            echo " Upgrading $FILE database to $1..."
+            echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+
+            # Disable exit on non 0
+            set +e
+
+            output="$(./t -t r -D -v -h $FILE)"
+            test_res=$?
+
+            # Enable exit on non 0
+            set -e
+
+            # Validate test result.
+            if [[ "$FILE" =~ "4.4."[0-6]"_unclean"$ ]]; then
+                if [[ "$test_res" == 0 ]]; then
+                    echo "$output"
+                    echo "Error: Upgrade of $FILE database to $1 succeeded!"
+                    echo "Databases generated with unclean shutdown from versions 4.4.[0-6] must fail!"
+                    exit 1
+                fi
+            elif [[ "$test_res" != 0 ]]; then
+                echo "$output"
+                echo "Error: Upgrade of $FILE database to $1 failed! Test result is $test_res."
+                exit 1
+            fi
+        done
+        cd $3
```

Comment:
I don't see a need of passing $3 just for cd operation, why can't we do it in the caller function?